<!doctype html>
<html >
<head>
    
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <!--[if lt IE 9]>
                <script src="http://css3-mediaqueries-js.googlecode.com/svn/trunk/css3-mediaqueries.js"></script>
        <![endif]-->
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta http-equiv="Content-Style-Type" content="text/css" />

    <!-- <link rel="stylesheet" type="text/css" href="template.css" /> -->
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/template.css" />

    <link href="https://vjs.zencdn.net/5.4.4/video-js.css" rel="stylesheet" />

    <script src="https://code.jquery.com/jquery-2.2.1.min.js"></script>
    <!-- <script type='text/javascript' src='menu/js/jquery.cookie.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.hoverIntent.minified.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.dcjqaccordion.2.7.min.js'></script> -->

    <!-- <link href="menu/css/skins/blue.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/graphite.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/grey.css" rel="stylesheet" type="text/css" /> -->
  
    <!-- <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script> -->
        
  
    <!-- <script src="script.js"></script> -->
  
    <!-- <script src="jquery.sticky-kit.js "></script> -->
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.cookie.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.hoverIntent.minified.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.dcjqaccordion.2.7.min.js'></script>

    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/blue.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/graphite.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/grey.css" rel="stylesheet" type="text/css" />
  
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/script.js"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/jquery.sticky-kit.js"></script>
    <meta name="generator" content="pandoc" />
  <title>README</title>
  <style type="text/css">code{white-space: pre;}</style>
  <style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
  </style>
</head>
<body>

    
    <div class="container">
    <div class="row">
            <div class="span12">

      
      <h1
id="pathwaycolormapper-powerpoint-based-visualization-of-gene-expression-data-on-pathway-diagrams">PathwayColorMapper:
PowerPoint-Based Visualization of Gene Expression Data on Pathway
Diagrams</h1>
<p><strong>PathwayColorMapper</strong> is a Python tool that allows
researchers to visualize gene-associated data, such as gene expression
levels, directly on pathway diagrams within PowerPoint files. This
enables intuitive, presentation-ready mapping of complex datasets using
customizable color scales.</p>
<h2 id="features">Features</h2>
<ul>
<li><strong>Custom Pathway Diagram Support</strong>: Annotate any
PowerPoint-based pathway diagram with your data.</li>
<li><strong>Color Mapping</strong>: Use gradient color scales to
represent numerical data such as upregulation and downregulation.</li>
<li><strong>Automated Processing</strong>: Automatically map data values
to diagram regions.</li>
<li><strong>PowerPoint Output</strong>: Create editable PowerPoint
slides for further customization and presentations.</li>
<li><strong>Flexible Data Input</strong>: Accepts standard data formats
like CSV or Excel for easy integration.</li>
</ul>
<hr />
<h2 id="project-structure">Project Structure</h2>
<pre><code>PathwayColorMapper/
├── README.md                     # Documentation with project overview and usage instructions
├── LICENSE.txt                   # MIT License file for the project
├── config/
│   └── config.yaml               # Configuration file defining color scales, ranges, and gradients
├── examples/
│   ├── example_data.csv          # Example gene expression data in CSV format (raw data)
│   ├── example_data_scaled.csv   # Example gene expression data in CSV format (Z-score normalized data)
│   ├── example_pathway.pptx      # Example PowerPoint file with a blank pathway diagram
│   ├── example_output.png        # Example image output showing annotated pathway (raw data)
│   ├── example_output_scaled.png # Example image output showing annotated pathway (scaled data)
├── output/                       # Directory for all generated outputs
│   ├── colorbar_vertical.png     # Generated vertical colorbar image
│   ├── colorbar_horizontal.png   # Generated horizontal colorbar image
│   ├── example_output.pptx       # Example annotated PowerPoint file (raw data)
│   ├── example_output_scaled.pptx# Example annotated PowerPoint file (scaled data)
├── scripts/
│   └── gene_pathway_plot.py      # Main Python script to annotate pathways with color-coded data
├── requirements.txt              # List of Python dependencies needed to run the project
├── .gitignore                    # Specifies files and directories to ignore in version control</code></pre>
<hr />
<h2 id="installation">Installation</h2>
<h3 id="requirements">Requirements</h3>
<ul>
<li>Python 3.8+</li>
<li>Libraries listed in <code>requirements.txt</code>.</li>
</ul>
<h3 id="steps">Steps</h3>
<ol type="1">
<li><p>Clone the repository:</p>
<div class="sourceCode" id="cb2"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">git</span> clone https://github.com/kyoshimr/PathwayColorMapper.git</span></code></pre></div></li>
<li><p>Navigate to the project directory:</p>
<div class="sourceCode" id="cb3"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> PathwayColorMapper</span></code></pre></div></li>
<li><p>Install the required dependencies:</p>
<div class="sourceCode" id="cb4"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pip</span> install <span class="at">-r</span> requirements.txt</span></code></pre></div></li>
</ol>
<hr />
<h2 id="usage">Usage</h2>
<h3 id="input-data-format">Input Data Format</h3>
<p>Prepare your data in a CSV or Excel file with the following
format:</p>
<table>
<thead>
<tr class="header">
<th>Gene</th>
<th>Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>TP53</td>
<td>2.5</td>
</tr>
<tr class="even">
<td>EGFR</td>
<td>-1.2</td>
</tr>
<tr class="odd">
<td>BRCA1</td>
<td>1.8</td>
</tr>
<tr class="even">
<td>MYC</td>
<td>0.5</td>
</tr>
<tr class="odd">
<td>PIK3CA</td>
<td>-0.8</td>
</tr>
</tbody>
</table>
<h3 id="steps-to-generate-pathway-visualization">Steps to Generate
Pathway Visualization</h3>
<ol type="1">
<li><p>Prepare your input files:</p>
<ul>
<li>Gene expression data in CSV or Excel format (see
<code>examples/example_data_scaled.csv</code>).</li>
<li>Pathway diagram in PowerPoint format (see
<code>examples/example_pathway.pptx</code>).</li>
</ul></li>
<li><p>Run the script:</p>
<div class="sourceCode" id="cb5"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> scripts/pathway_color_mapper.py <span class="at">--input</span> data.csv <span class="at">--pathway</span> pathway.pptx <span class="at">--output</span> output.pptx</span></code></pre></div></li>
<li><p>Open the output file:</p>
<ul>
<li><code>output.pptx</code> contains the annotated pathway with
color-coded genes.</li>
</ul></li>
</ol>
<hr />
<h2 id="example">Example</h2>
<h3 id="input-pathway-diagram">Input Pathway Diagram</h3>
<p>A blank or pre-designed pathway diagram in PowerPoint:</p>
<figure>
<img src="examples/example_input.png" alt="Example Input" />
<figcaption aria-hidden="true">Example Input</figcaption>
</figure>
<h3 id="output-pathway-diagram">Output Pathway Diagram</h3>
<p>Annotated pathway with a heatmap overlay:</p>
<figure>
<img src="examples/example_output_scaled.png" alt="Example Output" />
<figcaption aria-hidden="true">Example Output</figcaption>
</figure>
<hr />
<h2 id="example-data">Example Data</h2>
<p>Sample input files are provided in the <code>examples/</code>
directory to help you get started.</p>
<h3 id="files">Files</h3>
<ul>
<li><code>examples/example_data_scaled.csv</code>: A CSV file containing
sample gene expression data.</li>
<li><code>examples/example_data_scaled.xlsx</code>: An Excel version of
the same data.</li>
<li><code>examples/example_pathway.pptx</code>: A blank PowerPoint
pathway diagram.</li>
</ul>
<h3 id="how-to-use">How to Use</h3>
<p>Run the script with the provided sample data:</p>
<div class="sourceCode" id="cb6"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> scripts/pathway_color_mapper.py <span class="at">--input</span> examples/example_data_scaled.csv <span class="at">--pathway</span> examples/example_pathway.pptx <span class="at">--output</span> output/example_output_scaled.pptx</span></code></pre></div>
<p>This will generate an annotated pathway diagram in
<code>output/example_output_scaled.pptx</code>.</p>
<hr />
<h2 id="configuration">Configuration</h2>
<p>The behavior of the script is controlled by
<code>config/config.yaml</code>. Customize the following settings:</p>
<ul>
<li><strong>Color Scale</strong>: Define the gradient and range for
color mapping.</li>
<li><strong>Input/Output Settings</strong>: Adjust file paths and
additional parameters.</li>
</ul>
<p>Example <code>config.yaml</code> for Z-score normalized data:</p>
<div class="sourceCode" id="cb7"><pre
class="sourceCode yaml"><code class="sourceCode yaml"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">color_scale</span><span class="kw">:</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">min_value</span><span class="kw">:</span><span class="at"> </span><span class="fl">-2.0</span><span class="co">  # Minimum value for the color scale</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">max_value</span><span class="kw">:</span><span class="at"> </span><span class="fl">2.0</span><span class="co">   # Maximum value for the color scale</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">gradient</span><span class="kw">:</span><span class="co">        # Colors for the color scale</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="at">    </span><span class="kw">-</span><span class="at"> </span><span class="fu">color</span><span class="kw">:</span><span class="at"> blue</span><span class="co">  # Corresponds to negative values</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="at">    </span><span class="kw">-</span><span class="at"> </span><span class="fu">color</span><span class="kw">:</span><span class="at"> white</span><span class="co"> # Corresponds to mid values</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="at">    </span><span class="kw">-</span><span class="at"> </span><span class="fu">color</span><span class="kw">:</span><span class="at"> red</span><span class="co">   # Corresponds to positive values</span></span></code></pre></div>
<p>Example <code>config.yaml</code> for raw data (e.g., TPM):</p>
<div class="sourceCode" id="cb8"><pre
class="sourceCode yaml"><code class="sourceCode yaml"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">color_scale</span><span class="kw">:</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">min_value</span><span class="kw">:</span><span class="at"> </span><span class="dv">0</span><span class="co">       # Minimum value for the color scale</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">max_value</span><span class="kw">:</span><span class="at"> </span><span class="fl">2.0</span><span class="co">     # Maximum value for the color scale</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">gradient</span><span class="kw">:</span><span class="at">  viridis</span><span class="co"> # Colors for the color scale</span></span></code></pre></div>
<p>Mapped pathway with example raw data:</p>
<figure>
<img src="examples/example_output.png" alt="Example Output" />
<figcaption aria-hidden="true">Example Output</figcaption>
</figure>
<hr />
<h2 id="dependencies">Dependencies</h2>
<ul>
<li><code>pandas</code></li>
<li><code>matplotlib</code></li>
<li><code>python-pptx</code></li>
<li><code>pyyaml</code></li>
</ul>
<p>Install them using:</p>
<div class="sourceCode" id="cb9"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pip</span> install <span class="at">-r</span> requirements.txt</span></code></pre></div>
<hr />
<h2 id="contributing">Contributing</h2>
<p>Contributions are welcome! If you have suggestions for improvements,
encounter bugs, or want to add features, feel free to submit an issue or
pull request.</p>
<hr />
<h2 id="license">License</h2>
<p>This project is licensed under the MIT License. See the <a
href="LICENSE.txt">LICENSE</a> file for details.</p>
<hr />
<h2 id="acknowledgements">Acknowledgements</h2>
<p>PathwayColorMapper was created to simplify the visualization of
complex genomic datasets in a pathway context. Special thanks to the
open-source community for making tools like <code>python-pptx</code> and
<code>matplotlib</code> available for projects like this.</p>
            </div>
    </div>
  </div>
  <script src="https://vjs.zencdn.net/5.4.4/video.js"></script>

</body>
</html>
